# Data Preparation

In [2]:
# Basic Libraries
import numpy as np
import pandas as pd
import seaborn as sb
import matplotlib.pyplot as plt # we only need pyplot
sb.set() # set the default Seaborn style for graphics
pd.options.mode.chained_assignment = None  # default='warn'

In [3]:
# Importing data
df = pd.read_csv('data.csv')
df

Q1    Q2    Q3   Q4          Q4OTHER    Q5  \
0                 Employment Manager  11.0  17.0  1.0          Missing  17.0   
1             Assistant Prog Manager   2.0   2.0  3.0   technical writ  22.0   
2           Director of Laboratories   2.0   2.0  4.0          Missing  12.0   
3                     Project Leader  15.0  21.0  2.0          Missing  23.0   
4     Information Technology Anaylst   2.0   2.0  2.0          Missing   7.0   
..                               ...   ...   ...  ...              ...   ...   
751                 Internal Auditor   1.0   1.0  1.0          Missing   7.0   
752         Financial Representative   1.0   1.0  1.0          Missing   1.0   
753                       Accountant   1.0   1.0  1.0          Missing   5.0   
754         Systems Sales Consultant   1.0   1.0  9.0          Missing   1.0   
755                Software Engineer   1.0   1.0  2.0          Missing   7.0   

      Q6   Q7                        Q7A                    Q7B  ... CARIDEN4  \
0    2.0  1.0                   Teaching                Missing  ...      3.2   
1    NaN  2.0                    Missing                Missing  ...      5.6   
2    3.0  2.0                    Missing                Missing  ...      5.0   
3    2.0  1.0                   Military             Management  ...      1.0   
4    3.0  2.0                    Missing                Missing  ...      5.2   
..   ...  ...                        ...                    ...  ...      ...   
751  4.0  2.0                    Missing                Missing  ...      5.4   
752  1.0  1.0                   Teaching       Service Industry  ...      3.4   
753  3.0  1.0   restaurant manager/owner   construction manager  ...      4.5   
754  1.0  1.0                    Telecom                Defense  ...      4.4   
755  3.0  1.0                 Management              Financial  ...      4.0   

     ILEMP5  ILCF5  GENDER  Q4ITNON  ALLGRPS  MNONIT  FNONIT  ITMF  NONITMF  
0      2.67   2.50     1.0      0.0      3.0     NaN     0.0   NaN      1.0  
1      2.83   2.17     1.0      0.0      3.0     NaN     0.0   NaN      1.0  
2      1.67   1.00     0.0      0.0      1.0     0.0     NaN   NaN      0.0  
3      2.33   3.33     0.0      1.0      2.0     1.0     NaN   0.0      NaN  
4      3.50   2.33     0.0      1.0      2.0     1.0     NaN   0.0      NaN  
..      ...    ...     ...      ...      ...     ...     ...   ...      ...  
751    1.67   1.67     1.0      0.0      3.0     NaN     0.0   NaN      1.0  
752    3.17   2.00     0.0      0.0      1.0     0.0     NaN   NaN      0.0  
753    6.00   5.67     1.0      0.0      3.0     NaN     0.0   NaN      1.0  
754    2.50   2.33     0.0      0.0      1.0     0.0     NaN   NaN      0.0  
755    5.33   2.83     0.0      1.0      2.0     1.0     NaN   0.0      NaN  

[756 rows x 508 columns]

### Selecting relevant variables 

The dataset consists of responses from survey participants to 508 questions which reflects their job experience and their intentions to change employer.

24 out of the 508 variables will be used in creating models to predict employee's intention to change employer. 

### Description of the dataset from the study:

Accessed from: https://www.icpsr.umich.edu/web/ICPSR/studies/26782/variables

> **Q3** : Years with current employer 

> **Q17** : Have thought about looking for a job with a different employer in the past year (1-strongly disagree, 2-moderately disagree, 3-slightly disagree, 4-slightly agree, 5-moderately agree, 6-strongly agree)

> **Q44** : Current career choice was significantly influenced in a positive way by a high school teacher (1-strongly disagree, 2-moderately disagree, 3-slightly disagree, 4-slightly agree, 5-moderately agree, 6-strongly agree)

> **Q45** : Current career choice was significantly influenced in a positive way by a high school guidance counselor (1-strongly disagree, 2-moderately disagree, 3-slightly disagree, 4-slightly agree, 5-moderately agree, 6-strongly agree)

> **Q46** : Current career choice was significantly influenced in a positive way by my parents (1-strongly disagree, 2-moderately disagree, 3-slightly disagree, 4-slightly agree, 5-moderately agree, 6-strongly agree)

> **Q47** : Current career choice was significantly influenced in a positive way by my brother or sister (1-strongly disagree, 2-moderately disagree, 3-slightly disagree, 4-slightly agree, 5-moderately agree, 6-strongly agree)

> **Q48** : Current career choice was significantly influenced in a positive way by my spouse or partner (1-strongly disagree, 2-moderately disagree, 3-slightly disagree, 4-slightly agree, 5-moderately agree, 6-strongly agree)

> **Q49** : Current career choice was significantly influenced in a positive way by my children (1-strongly disagree, 2-moderately disagree, 3-slightly disagree, 4-slightly agree, 5-moderately agree, 6-strongly agree)

> **Q50** : Current career choice was significantly influenced in a positive way by a friend (1-strongly disagree, 2-moderately disagree, 3-slightly disagree, 4-slightly agree, 5-moderately agree, 6-strongly agree)

> **Q76** : Number of children in household

> **Q84** : Breakdown of household care and maintenance (0-no spouse or partner, 1-all my partner's responsibility, 2-mostly my partner's responsibility, 3-responsibility shared equally, 4-mostly my responsibility, 5-all my responsibility)

> **Q121** : Gender discrimination is a significant barrier to advancement in my career field (1-strongly disagree, 2-moderately disagree, 3-slightly disagree, 4-slightly agree, 5-moderately agree, 6-strongly agree)

> **Q123** : Discrimination on the basis of ethnic background is a significant barrier to advancement my career field (1-strongly disagree, 2-moderately disagree, 3-slightly disagree, 4-slightly agree, 5-moderately agree, 6-strongly agree)

> **Q140** : Have to miss family activities due to the amount of time I must spend on work responsibilities (1-strongly disagree, 2-moderately disagree, 3-slightly disagree, 4-slightly agree, 5-moderately agree, 6-strongly agree)

> **Q200** : Feel that he/she is being paid a fair amount for the work done (1-strongly disagree, 2-moderately disagree, 3-slightly disagree, 4-slightly agree, 5-moderately agree, 6-strongly agree)

> **Q205** : Benefits received are as good as most other organizations offer. (1-strongly disagree, 2-moderately disagree, 3-slightly disagree, 4-slightly agree, 5-moderately agree, 6-strongly agree)

> **Q212** : Enjoy coworkers (1-strongly disagree, 2-moderately disagree, 3-slightly disagree, 4-slightly agree, 5-moderately agree, 6-strongly agree)

> **DEMO4** : Education level (0-UNKNOWN CODE, 2-high school graduate or ged, 3-trade or technical training, 4-some college, no degree, 5-associate degree, 6-bachelor's degree, 7-master's degree, 8-professional degree, 9-doctorate)

> **WRKSTRS4** : Work stress (Overall Score using mean of 4 indicators)

> **AGE** : Age

> **JSSPROMO** : Satisfaction with promotion (undeclared range of scores, but higher scores denote higher satisfaction)

> **JSSOP** : Satisfaction with operating procedures (undeclared range of scores, but higher scores denote higher satisfaction)

> **JSSSUPER** : Satisfaction with supervision (undeclared range of scores, but higher scores denote higher satisfaction)

> **ILEMP5** : Intention to leave employer (overall score using mean of 5 indicators)


In [4]:
# Extracting relevant variables from the dataset 
extracted_df=df[['Q3', 'Q17', 'Q44', 'Q45', 'Q46', 'Q47', 'Q48', 'Q49', 'Q50', 'Q76', 'Q84', 'Q121', 'Q123', 'Q140', 'Q200', 'Q205', 'Q212', 'DEMO4', 'WRKSTRS4', 'AGE', 'JSSPROMO', 'JSSOP', 'JSSSUPER', 'ILEMP5']]
extracted_df

Q3  Q17  Q44  Q45  Q46  Q47  Q48  Q49  Q50  Q76  ...  Q200  Q205  Q212  \
0    17.0  4.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  2.0  ...   1.0   5.0   5.0   
1     2.0  6.0  1.0  1.0  6.0  2.0  1.0  1.0  2.0  0.0  ...   1.0   3.0   1.0   
2     2.0  1.0  5.0  2.0  5.0  1.0  1.0  1.0  5.0  0.0  ...   6.0   3.0   6.0   
3    21.0  2.0  NaN  NaN  NaN  NaN  NaN  NaN  NaN  0.0  ...   6.0   3.0   4.0   
4     2.0  4.0  2.0  1.0  2.0  5.0  4.0  4.0  NaN  0.0  ...   3.0   3.0   4.0   
..    ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...   ...   ...   ...   
751   1.0  1.0  1.0  1.0  4.0  5.0  1.0  1.0  5.0  0.0  ...   5.0   6.0   6.0   
752   1.0  4.0  1.0  1.0  3.0  1.0  3.0  1.0  1.0  0.0  ...   5.0   5.0   5.0   
753   1.0  6.0  1.0  1.0  5.0  1.0  5.0  5.0  5.0  1.0  ...   4.0   4.0   4.0   
754   1.0  4.0  1.0  1.0  1.0  1.0  4.0  2.0  2.0  2.0  ...   2.0   1.0   6.0   
755   1.0  6.0  1.0  1.0  4.0  5.0  5.0  1.0  5.0  1.0  ...   2.0   1.0   6.0   

     DEMO4  WRKSTRS4   AGE  JSSPROMO  JSSOP  JSSSUPER  ILEMP5  
0      6.0      1.50  42.0       8.0   10.0      12.0    2.67  
1      6.0      1.17  63.0       4.0    8.0       9.0    2.83  
2      9.0      1.83  35.0      10.0   11.0      12.0    1.67  
3      5.0      4.17  58.0       6.0   10.0      10.0    2.33  
4      4.0      4.83  27.0       8.0    4.0      10.0    3.50  
..     ...       ...   ...       ...    ...       ...     ...  
751    7.0      1.00  29.0      12.0   11.0      10.0    1.67  
752    6.0      2.50  24.0      10.0    7.0       6.0    3.17  
753    NaN      6.00  44.0       2.0    2.0       4.0    6.00  
754    7.0      4.00  37.0       6.0    6.0      10.0    2.50  
755    7.0      1.00  34.0       6.0   11.0       NaN    5.33  

[756 rows x 24 columns]

In [5]:
# Setting values which are 0 (unknown code) for 'DEMO4'
s=extracted_df.shape[0]
c=0;
while c<s:
    if extracted_df.at[c, 'DEMO4']==0:
        extracted_df.at[c, 'DEMO4']=np.nan
    c+=1

##### Removing & replacing null values 
- null values in 'Q17' column which will be used as a predictor variable for the model will be dropped to avoid 
misrepresentation of the data 
- filling the remaining null values with either median (numerical) or mode (categorical)

In [6]:
# dropping rows with null values in 'Q17' column 
new_df=extracted_df.dropna(subset=['Q17'])
new_df

Q3  Q17  Q44  Q45  Q46  Q47  Q48  Q49  Q50  Q76  ...  Q200  Q205  Q212  \
0    17.0  4.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  2.0  ...   1.0   5.0   5.0   
1     2.0  6.0  1.0  1.0  6.0  2.0  1.0  1.0  2.0  0.0  ...   1.0   3.0   1.0   
2     2.0  1.0  5.0  2.0  5.0  1.0  1.0  1.0  5.0  0.0  ...   6.0   3.0   6.0   
3    21.0  2.0  NaN  NaN  NaN  NaN  NaN  NaN  NaN  0.0  ...   6.0   3.0   4.0   
4     2.0  4.0  2.0  1.0  2.0  5.0  4.0  4.0  NaN  0.0  ...   3.0   3.0   4.0   
..    ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...   ...   ...   ...   
751   1.0  1.0  1.0  1.0  4.0  5.0  1.0  1.0  5.0  0.0  ...   5.0   6.0   6.0   
752   1.0  4.0  1.0  1.0  3.0  1.0  3.0  1.0  1.0  0.0  ...   5.0   5.0   5.0   
753   1.0  6.0  1.0  1.0  5.0  1.0  5.0  5.0  5.0  1.0  ...   4.0   4.0   4.0   
754   1.0  4.0  1.0  1.0  1.0  1.0  4.0  2.0  2.0  2.0  ...   2.0   1.0   6.0   
755   1.0  6.0  1.0  1.0  4.0  5.0  5.0  1.0  5.0  1.0  ...   2.0   1.0   6.0   

     DEMO4  WRKSTRS4   AGE  JSSPROMO  JSSOP  JSSSUPER  ILEMP5  
0      6.0      1.50  42.0       8.0   10.0      12.0    2.67  
1      6.0      1.17  63.0       4.0    8.0       9.0    2.83  
2      9.0      1.83  35.0      10.0   11.0      12.0    1.67  
3      5.0      4.17  58.0       6.0   10.0      10.0    2.33  
4      4.0      4.83  27.0       8.0    4.0      10.0    3.50  
..     ...       ...   ...       ...    ...       ...     ...  
751    7.0      1.00  29.0      12.0   11.0      10.0    1.67  
752    6.0      2.50  24.0      10.0    7.0       6.0    3.17  
753    NaN      6.00  44.0       2.0    2.0       4.0    6.00  
754    7.0      4.00  37.0       6.0    6.0      10.0    2.50  
755    7.0      1.00  34.0       6.0   11.0       NaN    5.33  

[753 rows x 24 columns]

In [7]:
mode=new_df.mode()

In [8]:
# replacing null with median for numerical data
array=['Q3', 'Q76', 'WRKSTRS4', 'AGE', 'JSSPROMO', 'JSSOP', 'JSSSUPER', 'ILEMP5']
for n in array:
    d=new_df[n].dropna()
    m=np.median(d)
    new_df[n].fillna(m, inplace = True)

In [9]:
# replacing null with mode for categorical data
array=['Q44', 'Q45', 'Q46', 'Q47', 'Q48', 'Q49', 'Q50', 'Q84', 'Q121', 'Q123', 'Q140', 'Q200', 'Q205', 'Q212', 'DEMO4']
for n in array:
    m=mode.at[0,str(n)]
    new_df[str(n)].fillna(m, inplace = True)

In [10]:
new_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 753 entries, 0 to 755
Data columns (total 24 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Q3        753 non-null    float64
 1   Q17       753 non-null    float64
 2   Q44       753 non-null    float64
 3   Q45       753 non-null    float64
 4   Q46       753 non-null    float64
 5   Q47       753 non-null    float64
 6   Q48       753 non-null    float64
 7   Q49       753 non-null    float64
 8   Q50       753 non-null    float64
 9   Q76       753 non-null    float64
 10  Q84       753 non-null    float64
 11  Q121      753 non-null    float64
 12  Q123      753 non-null    float64
 13  Q140      753 non-null    float64
 14  Q200      753 non-null    float64
 15  Q205      753 non-null    float64
 16  Q212      753 non-null    float64
 17  DEMO4     753 non-null    float64
 18  WRKSTRS4  753 non-null    float64
 19  AGE       753 non-null    float64
 20  JSSPROMO  753 non-null    float6

###### Creating a new variable 
- As our interest is in whether the presence of influence in career choice will affect an employee's intention to leave his/her employer, we will be converting responses for Q44-Q50 to either agree or disagree. If an employee agrees (option 4-6) to any of these questons, it would be considered as agree. Otherwise (option 1-3), it will be considered as disagree.

In [11]:
# creating the new variable
inf_cols=new_df[['Q44', 'Q45', 'Q46', 'Q47', 'Q48', 'Q49', 'Q50']]
def custom(row):
    if row.sum()>7*3:
        return "Agree"
    return "Disagree"
inf_col = pd.DataFrame(inf_cols.apply(custom, axis=1))

# concatenated df with new column
n_df=pd.concat([new_df, inf_col], axis=1)
n_df

Q3  Q17  Q44  Q45  Q46  Q47  Q48  Q49  Q50  Q76  ...  Q205  Q212  \
0    17.0  4.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  2.0  ...   5.0   5.0   
1     2.0  6.0  1.0  1.0  6.0  2.0  1.0  1.0  2.0  0.0  ...   3.0   1.0   
2     2.0  1.0  5.0  2.0  5.0  1.0  1.0  1.0  5.0  0.0  ...   3.0   6.0   
3    21.0  2.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  0.0  ...   3.0   4.0   
4     2.0  4.0  2.0  1.0  2.0  5.0  4.0  4.0  1.0  0.0  ...   3.0   4.0   
..    ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...   ...   ...   
751   1.0  1.0  1.0  1.0  4.0  5.0  1.0  1.0  5.0  0.0  ...   6.0   6.0   
752   1.0  4.0  1.0  1.0  3.0  1.0  3.0  1.0  1.0  0.0  ...   5.0   5.0   
753   1.0  6.0  1.0  1.0  5.0  1.0  5.0  5.0  5.0  1.0  ...   4.0   4.0   
754   1.0  4.0  1.0  1.0  1.0  1.0  4.0  2.0  2.0  2.0  ...   1.0   6.0   
755   1.0  6.0  1.0  1.0  4.0  5.0  5.0  1.0  5.0  1.0  ...   1.0   6.0   

     DEMO4  WRKSTRS4   AGE  JSSPROMO  JSSOP  JSSSUPER  ILEMP5         0  
0      6.0      1.50  42.0       8.0   10.0      12.0    2.67  Disagree  
1      6.0      1.17  63.0       4.0    8.0       9.0    2.83  Disagree  
2      9.0      1.83  35.0      10.0   11.0      12.0    1.67  Disagree  
3      5.0      4.17  58.0       6.0   10.0      10.0    2.33  Disagree  
4      4.0      4.83  27.0       8.0    4.0      10.0    3.50  Disagree  
..     ...       ...   ...       ...    ...       ...     ...       ...  
751    7.0      1.00  29.0      12.0   11.0      10.0    1.67  Disagree  
752    6.0      2.50  24.0      10.0    7.0       6.0    3.17  Disagree  
753    6.0      6.00  44.0       2.0    2.0       4.0    6.00     Agree  
754    7.0      4.00  37.0       6.0    6.0      10.0    2.50  Disagree  
755    7.0      1.00  34.0       6.0   11.0      10.0    5.33     Agree  

[753 rows x 25 columns]

In [12]:
# dropping the old variables
n_df.drop(['Q44', 'Q45', 'Q46', 'Q47', 'Q48', 'Q49', 'Q50'], axis=1, inplace=True)
n_df

Q3  Q17  Q76  Q84  Q121  Q123  Q140  Q200  Q205  Q212  DEMO4  WRKSTRS4  \
0    17.0  4.0  2.0  3.0   1.0   1.0   2.0   1.0   5.0   5.0    6.0      1.50   
1     2.0  6.0  0.0  4.0   1.0   1.0   6.0   1.0   3.0   1.0    6.0      1.17   
2     2.0  1.0  0.0  3.0   3.0   2.0   4.0   6.0   3.0   6.0    9.0      1.83   
3    21.0  2.0  0.0  5.0   1.0   1.0   2.0   6.0   3.0   4.0    5.0      4.17   
4     2.0  4.0  0.0  4.0   5.0   1.0   5.0   3.0   3.0   4.0    4.0      4.83   
..    ...  ...  ...  ...   ...   ...   ...   ...   ...   ...    ...       ...   
751   1.0  1.0  0.0  5.0   2.0   1.0   2.0   5.0   6.0   6.0    7.0      1.00   
752   1.0  4.0  0.0  2.0   1.0   1.0   5.0   5.0   5.0   5.0    6.0      2.50   
753   1.0  6.0  1.0  3.0   2.0   2.0   5.0   4.0   4.0   4.0    6.0      6.00   
754   1.0  4.0  2.0  3.0   1.0   1.0   6.0   2.0   1.0   6.0    7.0      4.00   
755   1.0  6.0  1.0  3.0   4.0   1.0   3.0   2.0   1.0   6.0    7.0      1.00   

      AGE  JSSPROMO  JSSOP  JSSSUPER  ILEMP5         0  
0    42.0       8.0   10.0      12.0    2.67  Disagree  
1    63.0       4.0    8.0       9.0    2.83  Disagree  
2    35.0      10.0   11.0      12.0    1.67  Disagree  
3    58.0       6.0   10.0      10.0    2.33  Disagree  
4    27.0       8.0    4.0      10.0    3.50  Disagree  
..    ...       ...    ...       ...     ...       ...  
751  29.0      12.0   11.0      10.0    1.67  Disagree  
752  24.0      10.0    7.0       6.0    3.17  Disagree  
753  44.0       2.0    2.0       4.0    6.00     Agree  
754  37.0       6.0    6.0      10.0    2.50  Disagree  
755  34.0       6.0   11.0      10.0    5.33     Agree  

[753 rows x 18 columns]

##### Converting variables with float data type to integer data type 
- removing the redundant decimal points for all variables, other than 'WRKSTRS4' variable, 'ILEMP5' and newly generated variable

In [13]:
# removing decimal points
float_cols=n_df.drop(['WRKSTRS4', 0, 'ILEMP5'], axis=1)
int_cols = pd.DataFrame(float_cols.applymap(int))
clean_df=pd.concat([int_cols, n_df[['WRKSTRS4', 0, 'ILEMP5']]], axis=1)
clean_df

Q3  Q17  Q76  Q84  Q121  Q123  Q140  Q200  Q205  Q212  DEMO4  AGE  \
0    17    4    2    3     1     1     2     1     5     5      6   42   
1     2    6    0    4     1     1     6     1     3     1      6   63   
2     2    1    0    3     3     2     4     6     3     6      9   35   
3    21    2    0    5     1     1     2     6     3     4      5   58   
4     2    4    0    4     5     1     5     3     3     4      4   27   
..   ..  ...  ...  ...   ...   ...   ...   ...   ...   ...    ...  ...   
751   1    1    0    5     2     1     2     5     6     6      7   29   
752   1    4    0    2     1     1     5     5     5     5      6   24   
753   1    6    1    3     2     2     5     4     4     4      6   44   
754   1    4    2    3     1     1     6     2     1     6      7   37   
755   1    6    1    3     4     1     3     2     1     6      7   34   

     JSSPROMO  JSSOP  JSSSUPER  WRKSTRS4         0  ILEMP5  
0           8     10        12      1.50  Disagree    2.67  
1           4      8         9      1.17  Disagree    2.83  
2          10     11        12      1.83  Disagree    1.67  
3           6     10        10      4.17  Disagree    2.33  
4           8      4        10      4.83  Disagree    3.50  
..        ...    ...       ...       ...       ...     ...  
751        12     11        10      1.00  Disagree    1.67  
752        10      7         6      2.50  Disagree    3.17  
753         2      2         4      6.00     Agree    6.00  
754         6      6        10      4.00  Disagree    2.50  
755         6     11        10      1.00     Agree    5.33  

[753 rows x 18 columns]

##### Renaming categories in categorical variables 

In [14]:
col1 = clean_df['Q84']

col1.replace(0, "No Partner", inplace=True)
col1.replace(1, "All Partner Responsibility", inplace=True)
col1.replace(2, "Mostly Partner Responsibility", inplace=True)
col1.replace(3, "Equal Responsibility", inplace=True)
col1.replace(4, "Mostly My Responsibility", inplace=True)
col1.replace(5, "All My Responsibility", inplace=True)

col2 = clean_df['DEMO4']

col2.replace(2, "High School Graduate or GED", inplace=True)
col2.replace(3, "Technical Training", inplace=True)
col2.replace(4, "Some college (No Degree)", inplace=True)
col2.replace(5, "Associate Degree", inplace=True)
col2.replace(6, "Bachelor Degree", inplace=True)
col2.replace(7, "Master Degree", inplace=True)
col2.replace(8, "Professional Degree", inplace=True)
col2.replace(9, "Doctorate", inplace=True)

col3 = clean_df['Q17']

col3.replace(1, "Disagree", inplace=True)
col3.replace(2, "Disagree", inplace=True)
col3.replace(3, "Disagree", inplace=True)
col3.replace(4, "Agree", inplace=True)
col3.replace(5, "Agree", inplace=True)
col3.replace(6, "Agree", inplace=True)

cols = clean_df.drop(['Q3', 'Q17', 'Q76', 'Q84', 'DEMO4','WRKSTRS4', 'AGE', 'JSSPROMO', 'JSSOP', 'JSSSUPER', 'ILEMP5', 0], axis=1)

cols.replace(1, "Strongly Disagree", inplace=True)
cols.replace(2, "Moderately Disagree", inplace=True)
cols.replace(3, "Slightly Disagree", inplace=True)
cols.replace(4, "Slightly Agree", inplace=True)
cols.replace(5, "Moderately Agree", inplace=True)
cols.replace(6, "Strongly Agree", inplace=True)
    
clean_df=pd.concat([col1, col2, col3, cols, clean_df[[0, 'Q3', 'Q76', 'WRKSTRS4', 'AGE', 'JSSPROMO', 'JSSOP', 'JSSSUPER', 'ILEMP5']]], axis=1)

clean_df

Q84                     DEMO4       Q17  \
0             Equal Responsibility           Bachelor Degree     Agree   
1         Mostly My Responsibility           Bachelor Degree     Agree   
2             Equal Responsibility                 Doctorate  Disagree   
3            All My Responsibility          Associate Degree  Disagree   
4         Mostly My Responsibility  Some college (No Degree)     Agree   
..                             ...                       ...       ...   
751          All My Responsibility             Master Degree  Disagree   
752  Mostly Partner Responsibility           Bachelor Degree     Agree   
753           Equal Responsibility           Bachelor Degree     Agree   
754           Equal Responsibility             Master Degree     Agree   
755           Equal Responsibility             Master Degree     Agree   

                    Q121                 Q123                 Q140  \
0      Strongly Disagree    Strongly Disagree  Moderately Disagree   
1      Strongly Disagree    Strongly Disagree       Strongly Agree   
2      Slightly Disagree  Moderately Disagree       Slightly Agree   
3      Strongly Disagree    Strongly Disagree  Moderately Disagree   
4       Moderately Agree    Strongly Disagree     Moderately Agree   
..                   ...                  ...                  ...   
751  Moderately Disagree    Strongly Disagree  Moderately Disagree   
752    Strongly Disagree    Strongly Disagree     Moderately Agree   
753  Moderately Disagree  Moderately Disagree     Moderately Agree   
754    Strongly Disagree    Strongly Disagree       Strongly Agree   
755       Slightly Agree    Strongly Disagree    Slightly Disagree   

                    Q200               Q205               Q212         0  Q3  \
0      Strongly Disagree   Moderately Agree   Moderately Agree  Disagree  17   
1      Strongly Disagree  Slightly Disagree  Strongly Disagree  Disagree   2   
2         Strongly Agree  Slightly Disagree     Strongly Agree  Disagree   2   
3         Strongly Agree  Slightly Disagree     Slightly Agree  Disagree  21   
4      Slightly Disagree  Slightly Disagree     Slightly Agree  Disagree   2   
..                   ...                ...                ...       ...  ..   
751     Moderately Agree     Strongly Agree     Strongly Agree  Disagree   1   
752     Moderately Agree   Moderately Agree   Moderately Agree  Disagree   1   
753       Slightly Agree     Slightly Agree     Slightly Agree     Agree   1   
754  Moderately Disagree  Strongly Disagree     Strongly Agree  Disagree   1   
755  Moderately Disagree  Strongly Disagree     Strongly Agree     Agree   1   

     Q76  WRKSTRS4  AGE  JSSPROMO  JSSOP  JSSSUPER  ILEMP5  
0      2      1.50   42         8     10        12    2.67  
1      0      1.17   63         4      8         9    2.83  
2      0      1.83   35        10     11        12    1.67  
3      0      4.17   58         6     10        10    2.33  
4      0      4.83   27         8      4        10    3.50  
..   ...       ...  ...       ...    ...       ...     ...  
751    0      1.00   29        12     11        10    1.67  
752    0      2.50   24        10      7         6    3.17  
753    1      6.00   44         2      2         4    6.00  
754    2      4.00   37         6      6        10    2.50  
755    1      1.00   34         6     11        10    5.33  

[753 rows x 18 columns]

##### Resetting index and renaming columns

In [15]:
# resetting index
clean_df=clean_df.reset_index(drop=True)
clean_df

Q84                     DEMO4       Q17  \
0             Equal Responsibility           Bachelor Degree     Agree   
1         Mostly My Responsibility           Bachelor Degree     Agree   
2             Equal Responsibility                 Doctorate  Disagree   
3            All My Responsibility          Associate Degree  Disagree   
4         Mostly My Responsibility  Some college (No Degree)     Agree   
..                             ...                       ...       ...   
748          All My Responsibility             Master Degree  Disagree   
749  Mostly Partner Responsibility           Bachelor Degree     Agree   
750           Equal Responsibility           Bachelor Degree     Agree   
751           Equal Responsibility             Master Degree     Agree   
752           Equal Responsibility             Master Degree     Agree   

                    Q121                 Q123                 Q140  \
0      Strongly Disagree    Strongly Disagree  Moderately Disagree   
1      Strongly Disagree    Strongly Disagree       Strongly Agree   
2      Slightly Disagree  Moderately Disagree       Slightly Agree   
3      Strongly Disagree    Strongly Disagree  Moderately Disagree   
4       Moderately Agree    Strongly Disagree     Moderately Agree   
..                   ...                  ...                  ...   
748  Moderately Disagree    Strongly Disagree  Moderately Disagree   
749    Strongly Disagree    Strongly Disagree     Moderately Agree   
750  Moderately Disagree  Moderately Disagree     Moderately Agree   
751    Strongly Disagree    Strongly Disagree       Strongly Agree   
752       Slightly Agree    Strongly Disagree    Slightly Disagree   

                    Q200               Q205               Q212         0  Q3  \
0      Strongly Disagree   Moderately Agree   Moderately Agree  Disagree  17   
1      Strongly Disagree  Slightly Disagree  Strongly Disagree  Disagree   2   
2         Strongly Agree  Slightly Disagree     Strongly Agree  Disagree   2   
3         Strongly Agree  Slightly Disagree     Slightly Agree  Disagree  21   
4      Slightly Disagree  Slightly Disagree     Slightly Agree  Disagree   2   
..                   ...                ...                ...       ...  ..   
748     Moderately Agree     Strongly Agree     Strongly Agree  Disagree   1   
749     Moderately Agree   Moderately Agree   Moderately Agree  Disagree   1   
750       Slightly Agree     Slightly Agree     Slightly Agree     Agree   1   
751  Moderately Disagree  Strongly Disagree     Strongly Agree  Disagree   1   
752  Moderately Disagree  Strongly Disagree     Strongly Agree     Agree   1   

     Q76  WRKSTRS4  AGE  JSSPROMO  JSSOP  JSSSUPER  ILEMP5  
0      2      1.50   42         8     10        12    2.67  
1      0      1.17   63         4      8         9    2.83  
2      0      1.83   35        10     11        12    1.67  
3      0      4.17   58         6     10        10    2.33  
4      0      4.83   27         8      4        10    3.50  
..   ...       ...  ...       ...    ...       ...     ...  
748    0      1.00   29        12     11        10    1.67  
749    0      2.50   24        10      7         6    3.17  
750    1      6.00   44         2      2         4    6.00  
751    2      4.00   37         6      6        10    2.50  
752    1      1.00   34         6     11        10    5.33  

[753 rows x 18 columns]

In [16]:
# Renaming columns
clean_df.set_axis(['HseCare', 'Edu', 'ChgEmp', 'GenDis', 'EthDis', 'MissFamAct', 'PaidFair', 'EmpBft', 'EnjCo', 'InfByOth', 'YrsCurEmp', 'NumChild', 'Stress', 'Age', 'Promo', 'OpProc', 'Supvr', 'IntLvEmp'], axis=1, inplace=True)
clean_df

HseCare                       Edu    ChgEmp  \
0             Equal Responsibility           Bachelor Degree     Agree   
1         Mostly My Responsibility           Bachelor Degree     Agree   
2             Equal Responsibility                 Doctorate  Disagree   
3            All My Responsibility          Associate Degree  Disagree   
4         Mostly My Responsibility  Some college (No Degree)     Agree   
..                             ...                       ...       ...   
748          All My Responsibility             Master Degree  Disagree   
749  Mostly Partner Responsibility           Bachelor Degree     Agree   
750           Equal Responsibility           Bachelor Degree     Agree   
751           Equal Responsibility             Master Degree     Agree   
752           Equal Responsibility             Master Degree     Agree   

                  GenDis               EthDis           MissFamAct  \
0      Strongly Disagree    Strongly Disagree  Moderately Disagree   
1      Strongly Disagree    Strongly Disagree       Strongly Agree   
2      Slightly Disagree  Moderately Disagree       Slightly Agree   
3      Strongly Disagree    Strongly Disagree  Moderately Disagree   
4       Moderately Agree    Strongly Disagree     Moderately Agree   
..                   ...                  ...                  ...   
748  Moderately Disagree    Strongly Disagree  Moderately Disagree   
749    Strongly Disagree    Strongly Disagree     Moderately Agree   
750  Moderately Disagree  Moderately Disagree     Moderately Agree   
751    Strongly Disagree    Strongly Disagree       Strongly Agree   
752       Slightly Agree    Strongly Disagree    Slightly Disagree   

                PaidFair             EmpBft              EnjCo  InfByOth  \
0      Strongly Disagree   Moderately Agree   Moderately Agree  Disagree   
1      Strongly Disagree  Slightly Disagree  Strongly Disagree  Disagree   
2         Strongly Agree  Slightly Disagree     Strongly Agree  Disagree   
3         Strongly Agree  Slightly Disagree     Slightly Agree  Disagree   
4      Slightly Disagree  Slightly Disagree     Slightly Agree  Disagree   
..                   ...                ...                ...       ...   
748     Moderately Agree     Strongly Agree     Strongly Agree  Disagree   
749     Moderately Agree   Moderately Agree   Moderately Agree  Disagree   
750       Slightly Agree     Slightly Agree     Slightly Agree     Agree   
751  Moderately Disagree  Strongly Disagree     Strongly Agree  Disagree   
752  Moderately Disagree  Strongly Disagree     Strongly Agree     Agree   

     YrsCurEmp  NumChild  Stress  Age  Promo  OpProc  Supvr  IntLvEmp  
0           17         2    1.50   42      8      10     12      2.67  
1            2         0    1.17   63      4       8      9      2.83  
2            2         0    1.83   35     10      11     12      1.67  
3           21         0    4.17   58      6      10     10      2.33  
4            2         0    4.83   27      8       4     10      3.50  
..         ...       ...     ...  ...    ...     ...    ...       ...  
748          1         0    1.00   29     12      11     10      1.67  
749          1         0    2.50   24     10       7      6      3.17  
750          1         1    6.00   44      2       2      4      6.00  
751          1         2    4.00   37      6       6     10      2.50  
752          1         1    1.00   34      6      11     10      5.33  

[753 rows x 18 columns]

### Updated description of data:

> **YrsCurEmp** : Years with current employer 

> **ChgEmp** : Have thought about looking for a job with a different employer in the past year 

> **NumChild** : Number of children in household

> **HseCare** : Breakdown of household care and maintenance 

> **GenDis** : Gender discrimination is a significant barrier to advancement in my career field 

> **EthDis** : Discrimination on the basis of ethnic background is a significant barrier to advancement my career field 

> **MissFamAct** : Have to miss family activities due to the amount of time I must spend on work responsibilities 

> **PaidFair** : Feel that he/she is being paid a fair amount for the work done 

> **EmpBft** : Benefits received are as good as most other organizations offer

> **EnjCo** : Enjoy coworkers 

> **Edu** : Education level 

> **Stress** : Work stress (Overall Score using mean of 4 indicators)

> **Age** : Age

> **Promo** : Satisfaction with promotion (undeclared range of scores, but higher scores denote higher satisfaction)

> **OpProc** : Satisfaction with operating procedures (undeclared range of scores, but higher scores denote higher satisfaction)

> **Supvr** : Satisfaction with supervision (undeclared range of scores, but higher scores denote higher satisfaction)

> **InfByOth** : Current career choice was significantly influenced in a positive way by others 

> **IntLvEmp** : Intention to leave employer (overall score using mean of 5 indicators)


In [17]:
# Converting dataframe to excel
clean_df.to_csv('clean_data.csv')